src/Haskell/Ide/Engine/Plugin/Ormolu.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Haskell.Ide.Engine.Plugin.Ormolu ( ormoluDescriptor ) where
+
+import Control.Monad.IO.Class ( liftIO, MonadIO(..) )
+import Control.Exception
+import Data.Aeson ( Value ( Null ) )
+import qualified Data.Text as T
+import Ormolu
+import Ormolu.Config (defaultConfig)
+import Ormolu.Exception (OrmoluException)
+import Haskell.Ide.Engine.MonadTypes
+import Haskell.Ide.Engine.PluginUtils
+
+ormoluDescriptor :: PluginId -> PluginDescriptor
+ormoluDescriptor plId = PluginDescriptor
+  { pluginId                 = plId
+  , pluginName               = "Ormolu"
+  , pluginDesc               = "A formatter for Haskell source code."
+  , pluginCommands           = []
+  , pluginCodeActionProvider = Nothing
+  , pluginDiagnosticProvider = Nothing
+  , pluginHoverProvider      = Nothing
+  , pluginSymbolProvider     = Nothing
+  , pluginFormattingProvider = Just provider
+  }
+
+provider :: FormattingProvider
+provider contents uri typ _opts =
+  case typ of 
+    -- // TODO Adequately throw an error on the following line
+    FormatRange _ -> return $ IdeResultFail (IdeError PluginError (T.pack $  "Selection formatting for Ormolu is not currently supported.") Null)
```

Comment:
I dont think we can actually throw an error here. 
If `ormolu` is selected as the formatter, `hsimport` tries to use its `FormatRange` type to format only the import line.
See the test cases `hsimport` with formatter to see an example when the formatter is used.

Reply:
We can just add a case for this in hsimport for now. But what we should really do is change the server capabilities on initialisation/configuration change to say that we don’t support range formatting anymore

Reply:
Specifically, we should probably take advantage of https://microsoft.github.io/language-server-protocol/specifications/specification-3-15/#client_unregisterCapability

Reply:
Can I have some more assistance on what would I need to do here? I mean, about checking `hsimport` and adding a case to it.

Reply:
I would say, dont worry about it. Register/unregister an capability is something to be done in haskell-lsp. Not sure if that is already implemented, but it is definitely out-of-scope for this pr. 
I would say, that a noop formatter and a log message is fine for this PR.